spring-cloud-netflix-eureka-client/src/main/java/org/springframework/cloud/netflix/eureka/config/DiscoveryClientOptionalArgsConfiguration.java

```diff
@@ -204,4 +207,56 @@ static class OnJerseyClientDisabled {
 
 	}
 
+	static class RestTemplateEnabledCondition extends AnyNestedCondition {
+
+		RestTemplateEnabledCondition() {
+			super(ConfigurationPhase.REGISTER_BEAN);
+		}
+
+		@Conditional(JerseyClientNotPresentOrNotEnabledCondition.class)
+		static class OnJerseyClientNotPresentOrNotEnabled {
+
+		}
+
+		@ConditionalOnProperty(prefix = "eureka.client", name = "webclient.enabled", matchIfMissing = true,
+				havingValue = "false")
+		static class OnWebClientDisabled {
+
+		}
+
+		@ConditionalOnProperty(prefix = "eureka.client", name = "restclient.enabled", matchIfMissing = true,
+				havingValue = "false")
```

Comment:
This should not be `matchIfMissing=true`. Since we now want the `RestClient` based implementation to be default, we only want to do `RestTemplate` if `RestClient` is explicitly disabled.

Reply:
I was really hoping that the default would change while working on this issue (it's too complicated).
Is this direction related to the issue below?
https://github.com/spring-cloud/spring-cloud-netflix/issues/4347

Reply:
Yes, we would like to change the default here to `RestClient`. That is why we don't want to use `RestTemplate` unless `RestClient` is explicitly disabled.